MIGRATE.md

```diff
@@ -0,0 +1,533 @@
+`snake_case`. For example, `client.runAction({ externalUserId: '...' })` becomes
+`client.actions.run({ external_user_id: '...' })`. This aligns to what's stated
+the [OpenAPI spec](https://api.pipedream.com/api-docs/swagger.json), but does
+**not apply** to the following arguments:
+
+1. Client constructor parameters like `clientId`, `clientSecret`, and
+   `projectId`.
+2. The props listed under `configured_props` (the naming follows whatever the
+   corresponding component defines).
+3. Request meta-arguments like `timeoutInSeconds`, `maxRetries`, and `headers`.
+
+### Migration examples
+
+#### Running actions
+
+##### v1.x (old)
+
+```javascript
+const result = await client.runAction({
+  externalUserId: 'jverce',
+  actionId: 'gitlab-list-commits',
+  configuredProps: {
+    gitlab: { authProvisionId: 'apn_kVh9AoD' },
+    projectId: 45672541,
+    refName: 'main',
+  },
+});
+```
+
+##### v2.x (new)
+
+```javascript
+const result = await client.actions.run({
+  external_user_id: 'jverce',
+  id: 'gitlab-list-commits',
+  configured_props: {
+    gitlab: { authProvisionId: 'apn_kVh9AoD' },
+    projectId: 45672541,
+    refName: 'main',
+  },
+});
+```
+
+#### Deploying triggers
+
+##### v1.x (old)
+
+```javascript
+const trigger = await client.deployTrigger({
+  externalUserId: 'jverce',
+  triggerId: 'gitlab-new-issue',
+  configuredProps: {
+    gitlab: {
+      authProvisionId: 'apn_kVh9AoD',
+    },
+    projectId: 45672541,
+  },
+  webhookUrl: 'https://events.example.com/gitlab-new-issue',
+});
+```
+
+##### v2.x (new)
+
+```javascript
+const trigger = await client.triggers.deploy({
+  external_user_id: 'jverce',
+  id: 'gitlab-new-issue',
+  configured_props: {
+    gitlab: {
+      authProvisionId: 'apn_kVh9AoD',
+    },
+    projectId: 45672541,
+  },
+  webhook_url: 'https://events.example.com/gitlab-new-issue',
+});
+```
+
+#### Managing accounts
+
+##### v1.x (old)
+
+```javascript
+// List accounts
+const accounts = await client.getAccounts({
+  external_user_id: 'jverce',
+  include_credentials: true,
+});
+
+// Get specific account
+const account = await client.getAccountById('apn_kVh9AoD', {
+  include_credentials: true,
+});
+```
+
+##### v2.x (new)
+
+```javascript
+// List accounts
+const accounts = await client.accounts.list({
+  external_user_id: 'jverce',
+  include_credentials: true,
+});
+
+// Get specific account
+const account = await client.accounts.retrieve('apn_kVh9AoD', {
+  include_credentials: true,
+});
+```
+
+#### Creating connect tokens
+
+##### v1.x (old)
+
+```javascript
+const token = await client.createConnectToken({
+  external_user_id: 'jverce',
+});
+```
+
+##### v2.x (new)
+
+```javascript
+const token = await client.tokens.create({
+  external_user_id: 'jverce',
+});
+```
+
+## Namespace mapping
+
+Here's a complete list of how v1.x methods map to v2.x namespaced methods:
+
+| v1.x Method               | v2.x Method                   |
```

Comment:
@jverce I think you're missing some methods, right? `configureComponent`, `reloadComponentProps` (and I'm looking for others)

Reply:
Hmm very likely, I just noticed that this is a **complete** list and not just a few examples.